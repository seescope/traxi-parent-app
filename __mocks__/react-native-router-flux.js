import React, { PropTypes } from 'react';
import { View } from 'react-native';

export const Actions = {
  createKid: jest.fn(),
  walkthrough: jest.fn(),
  congratulations: jest.fn(),
  thankyou: jest.fn(),
  areYouReady: jest.fn(),
  intro: jest.fn(),
};

export const Scene = (props) => <View {...props} />;

export const Router = ({ children }) => (
  <View>{children}</View>
);

Router.propTypes = {
  children: PropTypes.array.isRequired,
};
