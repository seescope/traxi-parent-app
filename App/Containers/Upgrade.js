// @flow
import React from 'react';
import { connect } from 'react-redux';
import { Image, Text, View, StyleSheet } from 'react-native';
import { MKButton } from 'react-native-material-kit';
import { Actions } from 'react-native-router-flux';

import { TRAXI_BLUE, GOOD, WHITE, GREY } from '../Constants/Colours';
import upgradeAccountAction from '../AsyncActions/UpgradeAccount';
import addAdditionalChild from '../AsyncActions/AddAdditionalChild';

import type { ParentAction } from '../Reducers/Parent';

type Props = {
  onPress: () => Promise<any>
};

type Dispatch = () => Promise<ParentAction>;

// $FlowFixMe
const LOGO = require('../Images/traxi_for_families.png');

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    flex: 1,
    backgroundColor: TRAXI_BLUE,
    paddingVertical: 64,
    paddingHorizontal: 32,
  },
  logo: {
    marginVertical: 32,
  },
  button: {
    marginHorizontal: 16,
  },
  buttonsContainer: {
    marginTop: 32,
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'row',
  },
  header: {
    fontFamily: 'Raleway-ExtraBold',
    fontSize: 35,
    color: WHITE,
    textAlign: 'center',
  },
  subHeader: {
    fontFamily: 'Raleway-Regular',
    fontSize: 19,
    color: WHITE,
    textAlign: 'center',
  },
});

type ButtonProps = {
  backgroundColour: string,
  textColour: string,
  onPress: () => {},
  text: string
};

const Button = (
  { backgroundColour, textColour, text, onPress }: ButtonProps
) => {
  const Component = MKButton.coloredButton()
    .withBackgroundColor(backgroundColour)
    .withTextStyle({
      fontFamily: 'Raleway-Bold',
      color: textColour,
    })
    .withText(text)
    .withOnPress(onPress)
    .build();

  return <View style={styles.button}><Component /></View>;
};

const Upgrade = ({ onPress }: Props) => (
  <View style={styles.container}>
    <Text style={styles.header}>Traxi for Families</Text>
    <Image resizeMode="contain" style={styles.logo} source={LOGO} />
    <Text style={styles.subHeader}>
      Monitor your family for just{'\n'} $1.99 per month.
    </Text>
    <View style={styles.buttonsContainer}>
      <Button
        backgroundColour={WHITE}
        textColour={GREY}
        onPress={() => Actions.pop()}
        text="Not yet"
      />
      <Button
        backgroundColour={GOOD}
        textColour={WHITE}
        onPress={onPress}
        text="Upgrade now"
      />
    </View>
  </View>
);

export const mapDispatchToProps = (dispatch: Dispatch): Props => ({
  onPress: async () => {
    await dispatch(upgradeAccountAction());
    await dispatch(addAdditionalChild());

    Actions.setName();
  },
});

export default connect(null, mapDispatchToProps, null)(Upgrade);
