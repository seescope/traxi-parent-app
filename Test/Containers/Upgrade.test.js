import React from 'react';
import renderer from 'react-test-renderer';
import { Actions } from 'react-native-router-flux';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
jest.mock('../../App/AsyncActions/UpgradeAccount', () =>
  jest.fn(() =>
    Promise.resolve({
      type: 'TEST_UPGRADE_ACCOUNT',
    })));
jest.mock('../../App/AsyncActions/AddAdditionalChild', () =>
  jest.fn(() =>
    Promise.resolve({
      type: 'TEST_ADD_ADDITIONAL_CHILD',
    })));

import Upgrade, { mapDispatchToProps } from '../../App/Containers/Upgrade';

const mockStore = configureStore([thunk]);
const testStore = mockStore({
  parentState: {
    name: 'Name',
    email: 'Email',
    kids: ['abc-123'],
  },
  kidsState: {
    'abc-123': {
      name: 'John Bobson',
    },
  },
  setupState: {
    kidUUID: 'abc-123',
  },
});

const UpgradeComponent = () => (
  <Provider store={testStore}>
    <Upgrade />
  </Provider>
);

it('renders the <Upgrade> component', () => {
  const tree = renderer.create(<UpgradeComponent />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('onPress', () => {
  test('Dispatches upgradeAccount, then addAdditionalChild and navigates to DeviceSetup', async () => {
    const MOCK_DISPATCH = jest.fn();
    const { onPress } = mapDispatchToProps(MOCK_DISPATCH);

    await onPress();

    expect(MOCK_DISPATCH).toHaveBeenCalledWith(
      expect.objectContaining({
        _65: {
          type: 'TEST_UPGRADE_ACCOUNT',
        },
      })
    );
    expect(MOCK_DISPATCH).toHaveBeenCalledWith(
      expect.objectContaining({
        _65: {
          type: 'TEST_ADD_ADDITIONAL_CHILD',
        },
      })
    );
    expect(Actions.setName).toHaveBeenCalled();
  });
});
