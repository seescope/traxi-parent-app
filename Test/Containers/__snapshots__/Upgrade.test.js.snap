// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the <Upgrade> component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#448AFF",
      "flex": 1,
      "paddingHorizontal": 32,
      "paddingVertical": 64,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#FFF",
        "fontFamily": "Raleway-ExtraBold",
        "fontSize": 35,
        "textAlign": "center",
      }
    }
  >
    Traxi for Families
  </Text>
  <Image
    resizeMode="contain"
    source={1}
    style={
      Object {
        "height": 667,
        "marginVertical": 32,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#FFF",
        "fontFamily": "Raleway-Regular",
        "fontSize": 19,
        "textAlign": "center",
      }
    }
  >
    Monitor your family for just
    
    
     $1.99 per month.
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 32,
      }
    }
  >
    <View
      style={
        Object {
          "marginHorizontal": 16,
        }
      }
    >
      <View
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "#FFF",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#545454",
              "fontFamily": "Raleway-Bold",
            }
          }
        >
          Not yet
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "marginHorizontal": 16,
        }
      }
    >
      <View
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "#4CAF50",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#FFF",
              "fontFamily": "Raleway-Bold",
            }
          }
        >
          Upgrade now
        </Text>
      </View>
    </View>
  </View>
</View>
`;
